processing/src/main/java/io/druid/segment/data/CompressedIntsIndexedSupplier.java

```diff
@@ -365,5 +366,13 @@ public void close() throws IOException
     {
       Closeables.close(holder, false);
     }
+
+    @Override
+    public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+    {
+      // ideally should inspect buffer, but at the moment of inspectRuntimeShape() call buffer might be null although
```

Comment:
So if we get those fields with subclasses different from the original object then we still get the first mapping and ignore the second implementation and eventually get unexpected/different results on dash, right?